### PR TITLE
Add io.github.MrPiggy92.ScriptedJourneys

### PR DESCRIPTION
**Pull request by MrPiggy92**
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly.
ScriptedJourneys is a text-based adventure game where players can explore various maps, solve puzzles, and embark on exciting quests. The game features rich storytelling and a variety of maps to choose from, offering a unique experience each time you play. Discover hidden secrets and unravel mysteries in this engaging adventure.

- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read the and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: io.github.MrPiggy92.ScriptedJourneys.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        "cp src/*.py /app/bin/",
+        "cp wrapper/wrapper.sh /app/bin/",
+        "chmod +x /app/bin/wrapper.sh",
+        "cp wrapper/first_run.sh /app/bin/",
+        "chmod +x /app/bin/first_run.sh",
+        "cp -r maps /app/lib",
+        "cp LICENSE /app/lib",
+        "cp wrapper/io.github.MrPiggy92.ScriptedJourneys.desktop /app/share/applications",
+        "cp wrapper/io.github.MrPiggy92.ScriptedJourneys.png /app/share/icons/hicolor/256x256/apps",
+        "cp wrapper/io.github.MrPiggy92.ScriptedJourneys.metainfo.xml /app/share/metainfo/"
```
Comment (hfiguiere) [inline io.github.MrPiggy92.ScriptedJourneys.json:40]:
```suggestion
        "install -Dm755 wrapper/wrapper.sh -t /app/bin/",
        "cp src/*.py /app/bin/",
        "install -Dm755 wrapper/first_run.sh -t /app/bin/",
        "install -Dm644 LICENSE -t /app/lib",
        "cp -r maps /app/lib",
        "install -Dm644 wrapper/io.github.MrPiggy92.ScriptedJourneys.desktop -t /app/share/applications",
        "install -Dm644 wrapper/io.github.MrPiggy92.ScriptedJourneys.png -t /app/share/icons/hicolor/256x256/apps",
        "install -Dm644 wrapper/io.github.MrPiggy92.ScriptedJourneys.metainfo.xml -t /app/share/metainfo/"
```

**Comment by hfiguiere** [comment]
bot, build io.github.MrPiggy92.ScriptedJourneys

**Comment by MrPiggy92** [comment]
bot, build io.github.MrPiggy92.ScriptedJourneys

**Comment by MrPiggy92** [comment]
Why has the build failed? The error output says developer id is missing, but the metainfo contains 
```
<developer id="io.github.MrPiggy92">
    <name>MrPiggy92</name>
  </developer>
```
I can't work out why the linter is outputting this. Is there a reason, and how should I fix it?

**Comment by bbhtt** [comment]
you aren't using the metainfo file here, you are using the one in https://github.com/MrPiggy92/ScriptedJourneys/blob/master/wrapper/io.github.MrPiggy92.ScriptedJourneys.metainfo.xml where it is missing.

Not sure why you have a duplicate copy here, merge the two and use the one in upstream repo.

**Comment by MrPiggy92** [comment]
I've updated it in the upstream repo, is that enough or should I do anything else before I rebuild?

File: ScriptedJourneys.flatpak
Comment (hfiguiere) [inline ScriptedJourneys.flatpak:1]: why is this here?

File: io.github.MrPiggy92.ScriptedJourneys.metainfo.xml
Comment (hfiguiere) [inline io.github.MrPiggy92.ScriptedJourneys.metainfo.xml:1]: you have it in the repository so remove this one.

File: ScriptedJourneys.flatpak
Comment (MrPiggy92) [reply ScriptedJourneys.flatpak:1]: I've never submitted to Flathub so I didn't know what I should put.

File: io.github.MrPiggy92.ScriptedJourneys.metainfo.xml
Comment (MrPiggy92) [reply io.github.MrPiggy92.ScriptedJourneys.metainfo.xml:1]: I have now

**Comment by MrPiggy92** [comment]
bot, build io.github.MrPiggy92.ScriptedJourneys

File: io.github.MrPiggy92.ScriptedJourneys.metainfo.xml
Comment (hfiguiere) [reply io.github.MrPiggy92.ScriptedJourneys.metainfo.xml:1]: no you didn't. you removed the manifest instead.

File: ScriptedJourneys.flatpak
Comment (hfiguiere) [reply ScriptedJourneys.flatpak:1]: there is never any mention of it anywhere not even in the doc.

**Comment by bbhtt** [comment]
You removed the manifest instead of removing the metainfo file.

**Comment by MrPiggy92** [comment]
bot, build io.github.MrPiggy92.ScriptedJourneys

File: io.github.MrPiggy92.ScriptedJourneys.json [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  ],
+  "modules": [
+    {
+      "name": "ScriptedJourneys",
+      "buildsystem": "simple",
+      "sources": [
+        {
+          "type": "git",
+          "url": "https://github.com/MrPiggy92/ScriptedJourneys.git",
+          "branch": "master",
```
Comment (bbhtt) [inline io.github.MrPiggy92.ScriptedJourneys.json:21]:
```suggestion
```

**Comment by MrPiggy92** [comment]
bot, build io.github.MrPiggy92.ScriptedJourneys

File: io.github.MrPiggy92.ScriptedJourneys.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{
+  "id": "io.github.MrPiggy92.ScriptedJourneys",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "22.08",
```
Comment (bbhtt) [inline io.github.MrPiggy92.ScriptedJourneys.json:4]: Can you use 24.08? 22.08 will be EOL by the end of this month.

File: io.github.MrPiggy92.ScriptedJourneys.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  "sdk": "org.freedesktop.Sdk",
+  "command": "wrapper.sh",
+  "finish-args": [
+    "--filesystem=home",
```
Comment (bbhtt) [inline io.github.MrPiggy92.ScriptedJourneys.json:8]: why is home needed?

Comment (MrPiggy92) [reply io.github.MrPiggy92.ScriptedJourneys.json:8]: It needs to make a config folder in .config. Should I specify the config folder that it will create on first run or just home/.config?

File: io.github.MrPiggy92.ScriptedJourneys.json (hunk 3 above)
Comment (MrPiggy92) [reply io.github.MrPiggy92.ScriptedJourneys.json:4]: when i change it to 24.08, it says Running appstream-compose
bwrap: execvp appstream-compose: No such file or directory
Error: ERROR: appstream-compose failed: Child process exited with code 1

File: io.github.MrPiggy92.ScriptedJourneys.json (hunk 4 above)
Comment (bbhtt) [reply io.github.MrPiggy92.ScriptedJourneys.json:8]: No it should use $XDG_CONFIG_HOME to create the config folder, then no permission is necessary.

File: io.github.MrPiggy92.ScriptedJourneys.json (hunk 3 above)
Comment (bbhtt) [reply io.github.MrPiggy92.ScriptedJourneys.json:4]: Can you do the change here so that we can test? That is an error on your end, you are likely missing appstreamcli package.

**Comment by MrPiggy92** [comment]
bot, build io.github.MrPiggy92.ScriptedJourneys
